Support importing components from the react-components package

> The issues go away if the components are imported from "theme"

Thanks, confirmed it helps. Updated the branch. There are some build errors with this approach:

```
[dts] src/generated/Accordion.ts(5,33): error TS7016: Could not find a declaration file for module '@vaadin/accordion/theme/lumo/vaadin-accordion.js'
```

Also, there are some components that are missing the `theme/lumo` entrypoint:

- `@vaadin/charts/theme/lumo/vaadin-chart-series.js`
- `@vaadin/icon/theme/lumo/vaadin-iconset.js`

--- a/package.json
+++ b/package.json
@@ -123,6 +123,7 @@
       "optional": true
     }
   },
+  "sideEffects": false,
   "devDependencies": {
     "@esm-bundle/chai": "^4.3.4-fix.0",
     "@testing-library/react": "^14.0.0",

--- a/package.json
+++ b/package.json
@@ -171,6 +171,10 @@
     "*.js.map"
   ],
   "exports": {
+    ".": {
+      "types": "./index.d.ts",
+      "default": "./index.js"
+    },
     "./package.json": "./package.json",
     "./Accordion.js": {
       "types": "./Accordion.d.ts",

--- a/scripts/generator.ts
+++ b/scripts/generator.ts
@@ -398,14 +398,13 @@ const elementFilesMap = await prepareElementFiles(descriptions);
 
 const sourceFiles = Array.from(elementFilesMap.entries(), ([element, data]) => generateReactComponent(element, data));
 
-function generateIndexDtsFile(elementNames: readonly string[]): SourceFile {
-  const sourceLines = [
-    `// Workaround for VSCode, enables TypeScript auto-imports form package.json`,
-    ...elementNames.map((elementName) => `import './${elementName}.js';`),
-  ];
+const elementNames = sourceFiles.map(({ fileName }) => basename(fileName, '.ts'));
+
+function generateIndexFile(elementNames: readonly string[], extension: string): SourceFile {
+  const sourceLines = [...elementNames.map((elementName) => `export * from './${elementName}.js';`)];
 
   return ts.createSourceFile(
-    resolve(rootDir, 'index.d.ts'),
+    resolve(rootDir, `index.${extension}`),
     sourceLines.join('\n'),
     ts.ScriptTarget.ES2019,
     undefined,
@@ -413,15 +412,9 @@ function generateIndexDtsFile(elementNames: readonly string[]): SourceFile {
   );
 }
 
-const indexDtsFile = generateIndexDtsFile(sourceFiles.map(({ fileName }) => basename(fileName, '.ts')));
+const indexDtsFile = generateIndexFile(elementNames, 'd.ts');
 
-const indexJsFile = ts.createSourceFile(
-  resolve(rootDir, 'index.js'),
-  '// Intentionally empty, please import from individual contained .js files',
-  ts.ScriptTarget.ES2019,
-  undefined,
-  ts.ScriptKind.JS,
-);
+const indexJsFile = generateIndexFile(elementNames, 'js');
 
 async function printAndWrite(file: SourceFile) {
   const contents = printer.printFile(file);

--- a/scripts/package-json-update.ts
+++ b/scripts/package-json-update.ts
@@ -9,6 +9,10 @@ const packageJsonPath = resolve(rootDir, 'package.json');
 const packageJson = JSON.parse(await readFile(packageJsonPath, 'utf8'));
 
 const exports: Record<string, PackageJson.Exports> = {
+  '.': {
+    types: './index.d.ts',
+    default: './index.js',
+  },
   './package.json': './package.json',
 };
 

